### PR TITLE
fix: LSDV-5368: add project_defined annotations to all columns

### PR DESCRIPTION
**Pull request by jombooth**
### PR fulfills these requirements
- [x] Commit message(s) and PR title follows the format `[fix|feat|ci|chore|doc]: TICKET-ID: Short description of change made` ex. `fix: DEV-XXXX: Removed inconsistent code usage causing intermittent errors`
- [x] Tests for the changes have been added/updated (for bug fixes/features)
- [ ] Docs have been added/updated (for bug fixes/features)
- [ ] Best efforts were made to ensure docs/code are concise and coherent (checked for spelling/grammatical errors, commented out code, debug logs etc.)
- [x] Self-reviewed and ran all changes on a local instance (for bug fixes/features)

#### Change has impacts in these area(s)
_(check all that apply)_
- [ ] Product design
- [ ] Backend (Database)
- [x] Backend (API)
- [x] Frontend

### Describe the reason for change
_(link to issue, supportive screenshots etc.)_

Introduces a notion of project defined columns, such that data manager columns defined in the project config can be distinguished from those which exist for all projects.

#### What does this fix?
_(if this is a bug fix)_

#### What is the new behavior?
_(if this is a breaking or feature change)_

#### What is the current behavior?
_(if this is a breaking or feature change)_

#### What libraries were added/updated?
_(list all with version changes)_

#### Does this change affect performance?
_(if so describe the impacts positive or negative)_

#### Does this change affect security?
_(if so describe the impacts positive or negative)_

#### What alternative approaches were there?
_(briefly list any if applicable)_

#### What feature flags were used to cover this change?
_(briefly list any if applicable)_

### Does this PR introduce a breaking change?
_(check only one)_
- [ ] Yes, and covered entirely by feature flag(s)
- [ ] Yes, and covered partially by feature flag(s)
- [ ] No
- [ ] Not sure (briefly explain the situation below)

### What level of testing was included in the change?
_(check all that apply)_
- [ ] e2e
- [ ] integration
- [ ] unit

### Which logical domain(s) does this change affect?
_(for bug fixes/features, be as precise as possible. ex. Authentication, Annotation History, Review Stream etc.)_

It affects the columns API (`api/dm/columns`), adding a new field.

### CODE REVIEW

**Comment by jombooth** [comment]
TODO: unit testing

**Comment by jombooth** [comment]
> TODO: unit testing

added in https://github.com/heartexlabs/label-studio/pull/4551/commits/25ef35ff398da7e4d4b7940a8df5eaf5bbddc193

File: label_studio/tests/data_manager/test_columns_api.py [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+    for c in columns:
+        assert 'project_defined' in c
+        assert c['project_defined'] is False
+
+
+def test_columns_api_annotates_config_defined_columns_with_project_defined_true(business_client):
+    config_with_text_column = (
+        """
+        <View>
+            <Text value="$text" name="artist" />
```
Comment (bmartel) [inline label_studio/tests/data_manager/test_columns_api.py:70]: Could we add a specific case for the original prevalent issue created with the `$data` usage as well.

**Review by bmartel — APPROVED** [review]

**Review by dredivaris — APPROVED** [review]
LGTM!

File: label_studio/tests/data_manager/test_columns_api.py (hunk 1 above)
Comment (jombooth) [reply label_studio/tests/data_manager/test_columns_api.py:70]: The original issue manifests only in LSE, it'll be specifically tested there (see also, how this change is used to fix the issue: https://github.com/HumanSignal/label-studio-enterprise/pull/4411/files#diff-a3d9c0b437f32cd5dca176d0a4268547afa904f30ec22fd7cee07ec09cf86130R18-R19)

**Pull request merged**